docs/conventions/component.md

```diff
@@ -0,0 +1,457 @@
+---
+title: Components coding conventions
+label: Components
+---
+
+## General rules
+
+In most cases, components are firstly done as generic components, then specialized for one or more systems.
+
+All components should follow the same file structure and naming conventions.
+
+If you are not familiar with the concept of systems in ECL, first have a look at [the system documentation](../systems).
+
+### Naming of components
+
+* Throughout the whole project, in `package.json` files, namespace should always be `@ecl/`, regardless whether work is to be done on a generic or system component. One reason is that `@ecl` namespace should be preserved for correct organization on [npmjs](https://www.npmjs.com/).
+* Flavored components should be prefixed with the name of their corresponding system and type (component, style, template): `[generic|ec|eu]-[component|style|template|utility]`.
+
+Example:
+```
+// Some full component name (in package.json)
```

Comment:
Just a personal opinion, this concept could be expressed more effectively by [tree](https://www.computerhope.com/unix/tree.htm) and some `// 👈 <------- use "hints"`